hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/OptUtils.java

```diff
@@ -172,31 +173,35 @@ public static boolean isPhysical(RelNode rel) {
     /**
      * Get possible physical rels from the given subset. Every returned input is guaranteed to have a unique trait set.
      *
-     * @param subset Subset.
+     * @param input Subset.
      * @return Physical rels.
      */
-    public static Collection<RelNode> getPhysicalRelsFromSubset(RelNode subset) {
-        if (subset instanceof RelSubset) {
-            RelSubset subset0 = (RelSubset) subset;
+    public static Collection<RelNode> getPhysicalRelsFromSubset(RelNode input) {
 
-            Set<RelTraitSet> traitSets = new HashSet<>();
+        RelNode convertedInput = convert(input, input.getTraitSet());
 
-            Set<RelNode> res = Collections.newSetFromMap(new IdentityHashMap<>());
+        if (!(convertedInput instanceof RelSubset)) {
+            return Collections.emptyList();
+        }
+
+        RelSubset logicalSubset = (RelSubset) convertedInput;
```

Comment:
Strictly speaking, `logicalSubset` is not necessarily logical. It could be physical as well. So I would rename this variable to avoid confusion.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18013/commits/acd00285d0965a1aec6627707f3845c33a22d223